modules/adagioBidAdapter.js

```diff
@@ -145,58 +148,38 @@ export const GlobalExchange = (function() {
   };
 })();
 
-export function adagioScriptFromLocalStorageCb(ls) {
-  try {
-    if (!ls) {
-      logWarn(`${LOG_PREFIX} script not found.`);
-      return;
-    }
-
-    const hashRgx = /^(\/\/ hash: (.+)\n)(.+\n)$/;
-
-    if (!hashRgx.test(ls)) {
-      logWarn(`${LOG_PREFIX} no hash found.`);
-      storage.removeDataFromLocalStorage(ADAGIO_LOCALSTORAGE_KEY);
-    } else {
-      const r = ls.match(hashRgx);
-      const hash = r[2];
-      const content = r[3];
+/**
+ * @param {string|number} adagiojsVersion The external adagio.js script version to use
+ * @returns {string} https://script.4dex.io/a[/VERSION]/adagio.js
+ */
+export function computeAdagioScriptUrl(adagiojsVersion) {
+  let versionDir = `/${ADAGIOJS_VERSION_LOCK}`;
+  const semverRgx = /^\d+\.\d+\.\d+$/;
 
-      if (verify(content, hash, ADAGIO_PUBKEY, ADAGIO_PUBKEY_E)) {
-        logInfo(`${LOG_PREFIX} start script.`);
-        Function(ls)(); // eslint-disable-line no-new-func
-      } else {
-        logWarn(`${LOG_PREFIX} invalid script found.`);
-        storage.removeDataFromLocalStorage(ADAGIO_LOCALSTORAGE_KEY);
-      }
-    }
-  } catch (err) {
-    logError(LOG_PREFIX, err);
+  if (adagiojsVersion === ADAGIOJS_VERSION_LATEST || semverRgx.test(adagiojsVersion)) {
+    versionDir = `/${adagiojsVersion}`;
+  } else {
+    logInfo(`${LOG_PREFIX} Invalid adagio.js version. Using the default version.`);
   }
+
+  return PUBLISHER_TAG_URL_TEMPLATE.replace(ADAGIOJS_VERSION_PLACEHOLDER, versionDir);
 }
 
 export function getAdagioScript() {
-  storage.getDataFromLocalStorage(ADAGIO_LOCALSTORAGE_KEY, (ls) => {
-    internal.adagioScriptFromLocalStorageCb(ls);
-  });
-
   storage.localStorageIsEnabled(isValid => {
-    if (isValid) {
-      loadExternalScript(ADAGIO_TAG_URL, BIDDER_CODE);
-    } else {
-      // Try-catch to avoid error when 3rd party cookies is disabled (e.g. in privacy mode)
-      try {
-        // ensure adagio removing for next time.
-        // It's an antipattern regarding the TCF2 enforcement logic
-        // but it's the only way to respect the user choice update.
-        window.localStorage.removeItem(ADAGIO_LOCALSTORAGE_KEY);
-        // Extra data from external script.
-        // This key is removed only if localStorage is not accessible.
-        window.localStorage.removeItem('adagio');
-      } catch (e) {
-        logInfo(`${LOG_PREFIX} unable to clear Adagio scripts from localstorage.`);
-      }
+    if (!isValid) {
+      return
+    }
+
+    const adagiojsVersion = bidderSettings.get('adagio', 'scriptVersion') || config.getConfig('adagio.scriptVersion');
+
+    // If the publisher explicitly set the version to 'none', we don't load anything.
+    if (adagiojsVersion === ADAGIOJS_VERSION_NONE) {
+      return;
     }
+
+    const url = computeAdagioScriptUrl(adagiojsVersion);
+    loadExternalScript(url, BIDDER_CODE, undefined, undefined, { id: `adagiojs-${getUniqueIdentifierStr()}`, version: adagiojsVersion });
```

Comment:
we'll need to discuss if we're going to let a bidder do this, likely we will not, as that was the initial plan. Why not distribute your sidecar script independently of prebid or make it another module?